File: 1주차/구민지/올바른 괄호.py
from collections import deque

def solution(s):
    stack=[]
    q=deque(s)
    while q:
        p=q.popleft()
        if len(stack)==0:
            stack.append(p)
            continue
        else:
            if stack[-1]=="(" and p==")":
                stack.pop()
            else:
                stack.append(p)
                
    if stack:
        return False
    return True

File: 1주차/김용상/올바른 괄호.java
import java.util.*;
class Solution {
    boolean solution(String s) {
        
        int n = s.length();
        Stack<Character>st = new Stack<>();
        for (int i=0;i<n;i++){
            if (s.charAt(i)=='(')
                st.push('(');
            else{
                if (st.size()==0) return false;
                st.pop();
            }
        }
        
        if (st.size()==0) return true;
        return false;
    }
}
